### PR TITLE
ci: publish-pr on external workflow

### PR DESCRIPTION
**Pull request by fpaul-1A**
## Proposed change

Currently, for external contributions, the workflow is not able to access secrets variables, so the job `publish-pr` is failing

One solution is to create another workflow triggered by `workflow_run`:`completed` that handle the publication

## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #(issue)

### CODE REVIEW

**Review by matthieu-crouzet — APPROVED** [review]

File: tools/github-actions/download-build-output/action.yml [hunk 1]
```diff
@@ -5,9 +5,33 @@ runs:
...
+      uses: actions/github-script@v6
+      if: github.event_name == 'workflow_run'
+      with:
+        script: |
+          let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+             owner: context.repo.owner,
+             repo: context.repo.repo,
+             run_id: context.payload.workflow_run.id,
+          });
+          let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
```
Comment (kpanot) [inline tools/github-actions/download-build-output/action.yml:23]: or
```typescript
const matchArtifact = allArtifacts.data.artifacts.find((artifact) => artifact.name == "dist");
```

File: tools/github-actions/download-build-output/action.yml [hunk 2]
```diff
@@ -5,9 +5,33 @@ runs:
...
+            return artifact.name == "dist"
+          })[0];
+          let download = await github.rest.actions.downloadArtifact({
```
Comment (kpanot) [inline tools/github-actions/download-build-output/action.yml:26]: as it is actually an exposed Github action, maybe you should handle le case of `matchArtifact == undefined` (and maybe provide an input for the `artifact name`)

File: tools/github-actions/download-build-output/action.yml [hunk 3]
```diff
@@ -5,9 +5,33 @@ runs:
...
         name: dist
         path: '.'
+
+    - name: 'Download artifact'
+      uses: actions/github-script@v6
+      if: github.event_name == 'workflow_run'
+      with:
+        script: |
+          let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+             owner: context.repo.owner,
```
[the remark this answers is not in the document]
Comment (fpaul-1A) [reply tools/github-actions/download-build-output/action.yml]: done

File: tools/github-actions/download-build-output/action.yml (hunk 1 above)
Comment (fpaul-1A) [reply tools/github-actions/download-build-output/action.yml]: done

File: tools/github-actions/download-build-output/action.yml (hunk 2 above)
Comment (fpaul-1A) [reply tools/github-actions/download-build-output/action.yml:31]: done

**Review by kpanot — APPROVED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

**Pull request merged**